selftests/lint.sh

```diff
@@ -28,3 +30,17 @@ ${PYLINT} ${PYLINT_OPTIONS} \
 	--disable=W0212,W0511,W0703,W0707,R,C,E1101,E1120,E0401,I0011 \
 	--enable=W1601,W1602,W1603,W1604,W1605,W1606,W1607,W1608,W1609,W1610,W1611,W1612,W1613,W1614,W1615,W1616,W1617,W1620,W1621,W1622,W1623,W1624,W1625,W1626,W1627,W1628,W1629,W1630,W1634,W1635,W1636,W1637,W1638,W1639,W1640,W1642,W1643,W1644,W1645,W1646,W1647,W1648,W1649,W1650,W1651,W1652,W1653,W1654,W1655,W1656,W1657,W1658,W1659,W1660,W1661,W1662 \
 	${FILES}
+status3=$?
+
+# Exit with error if one of the checks failed
+if [ $status1 -ne 0 ]; then
+	exit $status1
+fi
+
+if [ $status2 -ne 0 ]; then
+	exit $status2
+fi
+
+if [ $status3 -ne 0 ]; then
+	exit $status3
+fi
```

Comment:
While I like the idea of running all the tests and the general proposal, I don't find this very useful, because I will not be able to know where the exist code came from. And if multiple tests fail we are returning only one exitcode.  What about spiting this file into multiple lint-something.sh files?

Reply:
@beraldoleal Multiple lint-something files would lead to duplicate code. And maybe other changes in our CI.
 I understand you don't find useful not to know where the exit code come from, but that's the current behavior so no regression here. And what the CI needs to show are the problems detected.
The ideal fix here would be not having to call pylint 3 times, but that will take some time.

Reply:
> Multiple lint-something files would lead to duplicate code

Although this can be solved with a `source` call, indeed. I agree with you.

> I understand you don't find useful not to know where the exit code come from, but that's the current behavior so no regression here.

Currently, with the `#!/bin/sh -e` we have at least a basic consistency: the exit code of this script it is from the latest executed command. In case of failures, by looking at the output we can say where the exit code came from. So, unless I'm missing something, this implementation it is slightly different from the current one.

The way that I see this is if we would like to handle those as different tests, then either we split into different tests or be more verbose to keep it consistent and helpful, showing which command has failed. It is not just because the current implementation is "wrong", that we should keep introducing code that would lead contributors to a wrong interpretation of the results.

>  The ideal fix here would be not having to call pylint 3 times, but that will take some time.

I don't see as a problem to call it multiple times, since they are different types of checks/tests (specially the last case).

Reply:
Hi, 

Let me give an extended explanation of what I'm trying to improve and why this change doesn't make any regression with the exit codes.

The current `lint.sh` is messy and would need some refactoring but this is not the goal of this PR. The current script does the job(TM) .... with the exception that when one of the 3 checks fails, it'll exit. This is because the `set -e` in the top.
This behavior is annoying because when the CI runs the script and finds issues, it gives you a full list of all the issues and exists.
You're left believing you have a list of all the issues, only to find out there are probably more after you solve them!

With this change, you will run the 3 commands and get all the possible issues and then you'll get the same exit code you would have gotten with the current script. If the first command failed, you'll get this one. If the first and second one were ok but the third one failed, you'll get this last one. If all checks were OK, the script will exit with 0.

The only improvement I can think of is an echo message before running each of the commands if that helps to understand to what's going on. 
Besides that, I don't see what are the problems with this small PR. Especially how it could lead to a wrong interpretation of results, that's exactly what happens now!

Reply:
Hi @ana, thanks for explaining in details (appreciated), but I had understood the problem, and the proposed changed before.

>  With this change, you will run the 3 commands and get all the possible issues and then you'll get the same exit code you would have gotten with the current script. If the first command failed, you'll get this one. If the first and second one were ok but the third one failed, you'll get this last one. If all checks were OK, the script will exit with 0.

The problem is when more than one is failing. If the first and the second fails,  you are getting the exit code from the first *only*. And we miss the second error code. Or if all fails, we miss two codes, and give to users the impression that the result of the last command it as the first exit code. Before we missed the other errors, indeed! However, with regarding the exit code it was consistent: the exit code it was related to the latest command executed. What is not true here, and I find odd at least. 

(Note: just to be extra clear: I'm not blocking this, I left a comment exposing a inconsistency that I'm seeing here. I'm not nacking this because it is a small thing)

Reply:
> Thanks for the quick reply!
> 
> > The problem is when more than one is failing. If the first and the second fails, you are getting the exit code from the first _only_. And we miss the second error code. Or if all fails, we miss two codes, and give to users the impression that the result of the last command it was the first exit code. Before this PR, we missed the other errors, indeed! However, with regarding the exit code it was consistent: the exit code it was related to the latest command executed. What is not true here, and I find odd at least.
> 
> To be done in a new PR: Do you think a few lines explaining the results with the exit-codes in the screen before exiting the script help with this?

It would help, but it goes very much against what Avocado "preaches" as a best practice.  Every test, if at all possible, should be easily distinguishable in the results, there should be no need to go into the logs to find out which test failed.

Please take that into consideration on the new PR.

Reply:
> Do you think a few lines explaining the results with the exit-codes in the screen before exiting the script help with this?

It would help, but I would prefer to split the tests. Imagine the information lost at results.json|xml and any CI interface that parses those files.

Reply:
@beraldoleal ok, taking into account what @clebergnu said, please open an issue and we can discuss there.